modules/tensor_mechanics/include/materials/cohesive_zone_model/BiLinearMixedMode.h

```diff
@@ -0,0 +1,89 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "CZMComputeLocalTractionTotalBase.h"
+
+/**
+ * Implementation of the mixed mode bilinear traction separation law
+ * described in Mixed-Mode Decohesion Finite Elements for the Simulation of Delamination in
+ *Composite Materials, Pedro P. Camanho and Carlos G. Davila, NASA/TM-2002-211737
+ **/
+class BiLinearMixedMode : public CZMComputeLocalTractionTotalBase
+{
+public:
+  static InputParameters validParams();
+  BiLinearMixedMode(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties() override;
+
+  void computeInterfaceTractionAndDerivatives() override;
+
+  /// method computing the total traction
+  RealVectorValue computeTraction();
+
+  /// method computing the total traction derivatives w.r.t. the interface displacement jump
+  RankTwoTensor computeTractionDerivatives();
+
+  /// penalty elastic stiffness
+  const Real _stiffness;
+
+  /// damage variable
+  MaterialProperty<Real> & _d;
+
+  /// old damage variable value
+  const MaterialProperty<Real> & _d_old;
+
+  /// old interface displacement jump value
+  const MaterialProperty<RealVectorValue> & _interface_displacement_jump_old;
+
+  /// viscous regularization
+  const Real _viscosity;
+
+  /// critical Mode I fracture toughness
+  const Real _GI_C;
+
+  /// critical Mode II fracture toughness
+  const Real _GII_C;
+
+  /// onset normal seperation
+  const Real _delta_normal0;
+
+  /// onset shear seperation
+  const Real _delta_shear0;
+
+  /// effective displacement at damage initiation
+  MaterialProperty<Real> & _eff_disp_damage_init;
+
+  /// effective displacement at full degradation
+  MaterialProperty<Real> & _eff_disp_full_degradation;
```

Comment:
Those names are used in the cited paper, so I think it is fine. 

The reasons to make them material properties: 1. in case some users would like to lag them by using the old value. 2. We can output them to exodus file.